core/storage/feedback/gae_models.py

```diff
@@ -304,6 +304,32 @@ def get_messages(cls, exploration_id, thread_id):
             cls.thread_id == full_thread_id).fetch(feconf.DEFAULT_QUERY_LIMIT)
 
     @classmethod
+    def get_last_two_messages_of_threads(cls, exploration_ids, thread_ids):
+        """Returns a list of the last two messages of the given threads.
+
+        Args:
+            exploration_id: str. ID of the exploration the thread
+                belongs to.
+            thread_id: str. ID of the thread.
+
+        Returns:
+            list(int). A list of message ids present in the given thread, having
+                a maximum length of feconf.DEFAULT_QUERY_LIMIT.
+        """
+        full_thread_ids = [FeedbackThreadModel.generate_full_thread_id(
+            exploration_id, thread_id) for exploration_id, thread_id in zip(
+                exploration_ids, thread_ids)]
+
+        messages = []
+
+        for full_thread_id in full_thread_ids:
```

Comment:
You know, one thing that would speed this up is to just maintain a model that keeps track of the last two messages for each thread. Assuming that messages can't be deleted, this auxiliary model can be updated alongside the main model each time a new message comes in. Then you can just get by ID on the auxiliary model.

Thoughts?